src/main/java/seedu/address/logic/parser/PrefixDependentParser.java

```diff
@@ -0,0 +1,20 @@
+package seedu.address.logic.parser;
+
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.Command;
+
+/**
+ * Represents a {@code Parser} that makes use of Prefixes in {@link CliSyntax}.
+ */
+public abstract class PrefixDependentParser<T extends Command> implements Parser<T> {
```

Comment:
Hmm just a comment though: I found the name `PrefixDependentParser` a bit hard to decipher at first. Do you think something like `WithPrefixCommandParser` would be better if we were to make changes in the future?